### PR TITLE
Add rtloader build to agent build

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

- `invoke agent.build` call `invoke rtloader.build` and `invoke rtloader.install`.
- `invoke rtloader.clean` remove all artifacts
- `invoke agent.clean` call `invoke rtloader.clean`

### Additional Notes

> If you built an older version of the agent, you may have the error `make: *** No targets specified and no makefile found.  Stop.`.
  To solve the issue, you should clean the folder `rtloader` with the command `cd rtloader && git clean -fxd`. This command remove all git untracked files in `rtloader` folder!

**Can it be an issue for our customer?**

#### Files removed
CMake "out of the source build" is the standard way to use CMake and makes it easy to remove all artifacts.

New files removed by `rtloader.clean`:
* **dev/include/***
* **dev/lib/***
* CMakeFiles/
* Makefile
* cmake_install.cmake
* demo/CMakeFiles/
* demo/Makefile
* demo/cmake_install.cmake
* demo/demo
* install_manifest.txt
* rtloader/CMakeFiles/
* rtloader/Makefile
* rtloader/cmake_install.cmake
* rtloader/datadog-agent-rtloader.pc
* test/CMakeFiles/
* test/Makefile
* test/cmake_install.cmake
* three/CMakeFiles/
* three/Makefile
* three/cmake_install.cmake
* three/constants.h
* two/CMakeFiles/
* two/Makefile
* two/cmake_install.cmake
* two/constants.h
* two/libdatadog-agent-two.dylib

#### Performance
`inv agent.build` is 40% slower than before when there is nothing new (1.62s -> 2.24s).

The code https://github.com/DataDog/datadog-agent/pull/4788/files#diff-2d1c3a0be573aa65031befbdcff10e9eR98-R99 is 70% faster than calling `ctx.run("invoke rtloader.build")` and `ctx.run("invoke rtloader.install")`.

### CODE REVIEW

**Review by KSerrania — COMMENTED** [review]
Now that `rtloader.build` and `rtloader.install` are done by the `agent.build` command by default, the Agent software definitions should be changed to take this into account:
https://github.com/DataDog/datadog-agent/blob/5b8217ab440d4335d35de5820f8596f00417b7e6/omnibus/config/software/datadog-agent.rb#L61-L71
https://github.com/DataDog/datadog-agent/blob/5b8217ab440d4335d35de5820f8596f00417b7e6/omnibus/config/software/datadog-puppy.rb#L35
https://github.com/DataDog/datadog-agent/blob/5b8217ab440d4335d35de5820f8596f00417b7e6/omnibus/config/software/datadog-puppy.rb#L67

File: README.md [hunk 1]
```diff
@@ -46,25 +46,25 @@ To start working on the Agent, you can build the `master` branch:
...
 3. Install project's dependencies: `invoke deps`.
    Make sure that `$GOPATH/bin` is in your `$PATH` otherwise this step might fail.
-4. Build the `rtloader` dependency with `invoke rtloader.build && invoke
-   rtloader.install`. You will need CMake installed and a C++ compiler for this
-   to work.
-   `rtloader` is in charge of loading and running Python. By default `rtloader`
-   will be built for Python2, but you can choose which versions of Python you want
+4. Create a development `datadog.yaml` configuration file in `dev/dist/datadog.yaml`, containing a valid API key: `api_key: <API_KEY>`
+5. Build the agent with `invoke agent.build --build-exclude=systemd`. 
+   By default the agent will be built for Python3, but you can choose which versions of Python you want
```
Comment (remeh) [inline README.md]: typo: which version
Or, a proposal: `By default, the Agent will be built to use Python 3 but you can select which Python version you want to use`

File: tasks/rtloader.py [hunk 2]
```diff
@@ -10,6 +13,17 @@ def get_rtloader_path():
     here = os.path.abspath(os.path.dirname(__file__))
     return os.path.abspath(os.path.join(here, '..', 'rtloader'))
 
+def get_rtloader_build_path():
+    return os.path.join(get_rtloader_path(), 'build')
```
Comment (remeh) [inline tasks/rtloader.py]: 👌

File: tasks/rtloader.py [hunk 3]
```diff
@@ -56,43 +67,59 @@ def build(ctx, install_prefix=None, python_runtimes='3', cmake_options='', arch=
...
+
+    ctx.run("cd {} && cmake {} {}".format(rtloader_build_path, cmake_args, get_rtloader_path()))
+    run_make_command(ctx)
 
 @task
 def clean(ctx):
     """
     Clean up CMake's cache.
     Necessary when the paths to some libraries found by CMake (for example Python) have changed on the system.
     This command does not clean all the temporary artifacts created by CMake.
```
Comment (remeh) [inline tasks/rtloader.py]: This line could be removed.

**Review by remeh — DISMISSED** [review]
LGTM! Made some small feedback.

File: README.md [hunk 4]
```diff
@@ -46,25 +46,25 @@ To start working on the Agent, you can build the `master` branch:
...
    virtualenvs): `invoke agent.build
+   --python-runtimes 2,3
    --python-home-2=$GOPATH/src/github.com/DataDog/datadog-agent/venv2
    --python-home-3=$GOPATH/src/github.com/DataDog/datadog-agent/venv3`.
   Running `invoke agent.build`:
     * Discards any changes done in `bin/agent/dist`.
     * Builds the Agent and writes the binary to `bin/agent/agent`.
     * Copies files from `dev/dist` to `bin/agent/dist`. See `https://github.com/DataDog/datadog-agent/blob/master/dev/dist/README.md` for more information.
+  If you built an older version of the agent, you may have the error `make: *** No targets specified and no makefile found.  Stop.`.
+  To solve the issue, you should clean the folder `rtloader` with the command `cd rtloader && git clean -fxd`. This command remove all git untracked files in `rtloader` folder!
```
[the remark this answers is not in the document]
Comment (ogaca-dd) [reply README.md]: Fixed

File: tasks/agent.py [hunk 5]
```diff
@@ -434,6 +440,8 @@ def clean(ctx):
     print("Remove agent binary folder")
     ctx.run("rm -rf ./bin/agent")
 
+    print("Cleaning rtloader")
+    rtloader_clean(ctx)
```
Comment (KSerrania) [inline tasks/agent.py]: Should we only clean when `exclude_rtloader` (or `puppy`) is not set, to make the build behave exactly the same as before when `exclude_rtloader` / `puppy` is set?

Comment (ogaca-dd) [reply tasks/agent.py]: We do not have the arguments `exclude_rtloader` (or `puppy`) in `clean` function and I do no think it worth adding them.

**Review by remeh — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

**Pull request merged**